### PR TITLE
Draft for JOSS publication

### LINKED ISSUE #57: Consistent naming schema for data wrangling and transformation functions
https://easystats.github.io/datawizard/reference/index.html

due to aliases for some functions that start with `data_`, and pkgdown.yml using `starts_with("data")`, functions are shown under two headers.

### LINKED ISSUE #59: Writing a JOSS paper
Once the API is more stable (which should be the case in a few months), we should definitely do this.

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Closes #59 

Checklist:

- [x] Check if the tables are necessary. If we can include something in #87, we can leave out the two tables.
- [x] Make sure `describe_distribution()` output doesn't overflow in PDF output
- [x] Resolve #57 and #197
- [ ] ~~Make a new CRAN release~~ (irrelevant to the creation of draft itself)

### CODE REVIEW

File: paper/paper.Rmd
Comment (IndrajeetPatil) [inline paper/paper.Rmd:96]: I am deliberately not being comprehensive because we might add more functions in the future and `...` cover all of those.

File: paper/paper.Rmd
Comment (IndrajeetPatil) [inline paper/paper.Rmd:132]: I am deliberately not being comprehensive because we might add more functions in the future and `...` cover all of those.

File: paper/paper.Rmd
Comment (IndrajeetPatil) [inline paper/paper.Rmd:2]: I am not fully sold on the terms I am using here: Data Wrangling and Data Transformations.

Basically, we should distinguish between two key categories in which _datawizard_ functions can be broken down:
one that deal with wrangling and the other that deal with statistical transformations. 

But I am not sure what other term to use here, or should we just give up on trying to force the section to have "data" prefix?

In that case, this would be: 
> datawizard: An R Package for Easy Data Wrangling and Statistical Transformations

File: paper/paper.Rmd
Comment (IndrajeetPatil) [inline paper/paper.Rmd:147]: Not sure if a table is necessary here. We don't have many functions.

**Comment by bwiernik** [comment]
I would characterize the offerings of datawizard a little differently:

- data manipulation and summarization

Basically, we offer 

1. Lightweight alternatives to tidyverse mutate/pivot/rename offerings
2. Stats-y functions for data manipulation (centering, scaling, etc)
3. Lightweight summary table (table 1) functions

**Comment by mattansb** [comment]
I tend to agree with @bwiernik.

Perhaps we can separate functions that are intended to manipulate a data frame from functions intended to transform the values of a vector (and optionally all vectors in a data frame).

File: paper/paper.Rmd
Comment (etiennebacher) [inline paper/paper.Rmd:151]: add an example to show that it also works with a full dataset?

File: paper/paper.Rmd
Comment (etiennebacher) [inline paper/paper.Rmd:155]: we need to add a license to the repo

File: paper/paper.Rmd
Comment (IndrajeetPatil) [inline paper/paper.Rmd:21]: Can you please also add this to `DESCRIPTION` file when you make a PR for #188

Comment (etiennebacher) [reply paper/paper.Rmd:21]: yes I will

**Comment by strengejacke** [comment]
I don't think I should be first author of every JOSS paper... I mean, I don't mind, but maybe someone else could also be first author here?

File: paper/paper.Rmd
Comment (IndrajeetPatil) [reply paper/paper.Rmd:155]: Done.

File: paper/paper.Rmd
Comment (IndrajeetPatil) [reply paper/paper.Rmd:151]: Done.

But the table output overflows in PDF; we will need to deal with that somehow.

File: paper/paper.Rmd
Comment (IndrajeetPatil) [reply paper/paper.Rmd]: More common in DS literature, but, you are right, _variable_ is better.

**Comment by IndrajeetPatil** [comment]
If you want to read the PDF for the draft, it's [here](https://github.com/easystats/datawizard/blob/59_joss_paper/paper/paper.pdf).

@bwiernik I think I am already following the three-point structure you outline in your comment.

**Comment by DominiqueMakowski** [comment]
I reordered the intro a bit so that the logic easystats -> lightweight utility -> it's more than that   is clearer

File: paper/paper.Rmd
Comment (DominiqueMakowski) [inline paper/paper.Rmd:2]: I would call it "Data Preparation", it encompasses wrangling + transfo + implies that it's pre-analysis

Comment (IndrajeetPatil) [reply paper/paper.Rmd]: Good point. Done!

**Comment by etiennebacher** [comment]
@IndrajeetPatil I tried to solve the problem with the `describe_distribution()` output by creating a table. It's not the raw output of the function but at least it fits in the pdf. I couldn't find a solution with the raw output, maybe going back to `describe_distribution(mtcars$wt)` is better

**Comment by IndrajeetPatil** [comment]
Thanks, Etienne! Looks much better now.

Screenshot 2022-07-11 at 14 56 57

**Comment by IndrajeetPatil** [comment]
Okay, I think this is ready for another round of review from everyone. 

Let me know if there are still any other changes that you would like to make. Once CRAN reopens for submission, I would like to get the new version on CRAN, and then submit the paper.

**Review by bwiernik — CHANGES_REQUESTED** [review]

File: paper/paper.Rmd
Comment (bwiernik) [inline paper/paper.Rmd:34]:
```suggestion
  name: Independent Researcher
```

Comment (bwiernik) [reply paper/paper.Rmd:34]: Add an author note for me: "Brenton Wiernik is currently an independent researcher and Research Scientist at Meta,  Demography and Survey Science. The current research was in an independent capacity."

**Review by bwiernik — CHANGES_REQUESTED** [review]

**Comment by bwiernik** [comment]
Ah sorry grabbed the wrong file

**Comment by mattansb** [comment]
(Sorry for being less responsive on this, still recovering from covid... Don't wait for me)

File: paper/paper.Rmd
Comment (etiennebacher) [inline paper/paper.Rmd:80]: "reshape data" -> "some columns need to be reshaped"?

File: paper/paper.Rmd
Comment (etiennebacher) [inline paper/paper.Rmd:161]: "at Github" -> "on Github"?

File: paper/paper.Rmd
Comment (etiennebacher) [inline paper/paper.Rmd:165]: maybe link to https://easystats.github.io/easystats/ instead

**Comment by etiennebacher** [comment]
I added a few comments but otherwise I'm ok with the current version.

**Review by mattansb — APPROVED** [review]

**Review by etiennebacher — APPROVED** [review]

**Comment by IndrajeetPatil** [comment]
Waiting for your reviews/approvals now: @bwiernik, @DominiqueMakowski, and @strengejacke

File: paper/paper.Rmd
Comment (bwiernik) [inline paper/paper.Rmd:63]:
```suggestion
The `{datawizard}` package for the R programming language [@base2021] provides a lightweight toolbox to assist in keys steps involved in any data analysis workflow: (1) wrangling the raw data to get it in the needed form, (2) applying preprocessing steps and statistical transformations, and (3) compute statistical summaries of data properties and distributions. Therefore, it can be a valuable tool for R users and developers looking for a lightweight option for data preparation.
```

File: paper/paper.Rmd
Comment (bwiernik) [inline paper/paper.Rmd:70]:
```suggestion
Because `{datawizard}` is also meant to be used and adopted easily by a wide range of users, its workflow and syntax are designed to be similar to `{tidyverse}` (@Wickham2019), a widely used ecosystem of R packages. Thus, users familiar with the `{tidyverse}` can easily translate their knowledge and make full use of `{datawizard}`.
```

File: paper/paper.Rmd
Comment (bwiernik) [inline paper/paper.Rmd:74]:
```suggestion
Lastly, `{datawizard}` also provides a toolbox to create a detailed profile of data properties. This is a common step in data analysis, but it is not readily available in base R or many modeling packages, so its inclusion makes `{datawizard}` a one-stop-shop for data preparation tasks.
```

File: paper/paper.Rmd
Comment (bwiernik) [inline paper/paper.Rmd:58]:
```suggestion
library(datawizard)
set.seed(2016)
```

File: paper/paper.Rmd
Comment (bwiernik) [inline paper/paper.Rmd:74]:
```suggestion
Lastly, `{datawizard}` also provides a toolbox to create a detailed summaries of data properties and distributions (e.g., tables of descriptive statistics for each variable). This is a common step in data analysis, but it is not available in base R or many modeling packages, so its inclusion makes `{datawizard}` a one-stop-shop for data preparation tasks.
```

File: paper/paper.Rmd
Comment (bwiernik) [inline paper/paper.Rmd:144]:
```suggestion
## Summaries of Data Properties and Distributions
```

**Review by bwiernik — APPROVED** [review]

File: paper/paper.Rmd
Comment (bwiernik) [inline paper/paper.Rmd:75]:
```suggestion
Lastly, `{datawizard}` also provides a toolbox to create detailed summaries of data properties and distributions (e.g., tables of descriptive statistics for each variable). This is a common step in data analysis, but it is not available in base R or many modeling packages, so its inclusion makes `{datawizard}` a one-stop-shop for data preparation tasks.
```

**Pull request merged**
